packages/xstate-react/README.md

```diff
@@ -51,9 +51,9 @@ A [React hook](https://reactjs.org/hooks) that interprets the given `machine` an
 - `machine` - An [XState machine](https://xstate.js.org/docs/guides/machines.html).
 - `options` (optional) - [Interpreter options](https://xstate.js.org/docs/guides/interpretation.html#options) OR one of the following Machine Config options: `guards`, `actions`, `activities`, `services`, `delays`, `immediate`, `context`, or `state`.
 
-**Returns** a tuple of `[current, send, service]`:
+**Returns** a tuple of `[state, send, service]`:
 
-- `current` - Represents the current state of the machine as an XState `State` object.
+- `state` - Represents the current state of the machine as an XState `State` object.
```

Comment:
`state` name seems to be a little bit overloaded, might mean different things - currently it is an instance of `State` though, so `state` seems to be an appropriate name for this, but is also a little bit ambiguous, can we think of something better? maybe `State` should be renamed to something else in the next major?

Reply:
I don't think it's ambiguous - it's an object that represents the overall state.

Reply:
Maybe I was overthinking this, I think you are right - the name is OK. One thing that I would love exploring in the future (it's a minor pet peeve of mine) is removing `event` & `actions` from this object. I don't think those are much needed for it as they are really temporary. Removing them would allow us to only create new States when something actually changes (so it would improve perf slightly). Either way - this definitely is just an idea to explore for me right now, not something that I want to jump into in the nearest future.

Reply:
Those are necessary for the interpreter to properly decide what to do next - actions are necessary so the interpreter knows what to execute, and the event is necessary for action implementations that require both the `context` and `event`. I depend on them both in multiple projects.

Reply:
I understand why both of those are there, but maybe some alternative API for returning those could be explored. I hadn't given this that much thought yet, it's just something that I have noticed as potential improvement, but definitely not without exploring alternatives first. Might even in the end come to conclusion that it's fine as is

Reply:
Re: alternative API, it's important (IMO) to consider all the use-cases for this. For example, if this is a state you receive as a JSON payload in response to a GET/POST request to a remote statechart service, you ideally want all that information up-front in that JSON object.